libsolidity/codegen/ExpressionCompiler.cpp

```diff
@@ -410,6 +410,43 @@ bool ExpressionCompiler::visit(TupleExpression const& _tuple)
 bool ExpressionCompiler::visit(UnaryOperation const& _unaryOperation)
 {
 	CompilerContext::LocationSetter locationSetter(m_context, _unaryOperation);
+
+	if (_unaryOperation.annotation().userDefinedFunction.set())
+	{
+		FunctionDefinition const* function = *_unaryOperation.annotation().userDefinedFunction;
```

Comment:
Needs an assert that the definition is not null. Same for binary and in IR generator.